### PR TITLE
FIX: os.path.join does not like it when the second path has a `/`

### PR DESCRIPTION
**Pull request by CJ-Wright**
in the starting position, we `lstrip('/')` just in case.

### CODE REVIEW

**Comment by CJ-Wright** [comment]
@tacaswell This fixes the `FileNotFoundError` I was having.

**Comment by tacaswell** [comment]
This is too blunt of a fix.  We should check if the rpath is absolute and if is _and_ the root is empty, set root to `os.sep` and pull it off the left of the `rpath`.

**Comment by CJ-Wright** [comment]
Yes but that doesn't fix my problem.

**Comment by tacaswell** [comment]
you should never have both an absolute rpath _and_ a non-trivial root.

**Comment by CJ-Wright** [comment]
Yea well that didn't happen. 
The root was `'/home/deployment/mongo_files'` and the rpath was `/direct/XF28ID1/pe1_data/2017/03/25/'`

**Comment by CJ-Wright** [comment]
I think that some of this has do do with XPD's lack of a root on many of its filestore records.

File: filestore/fs.py [hunk 1]
```diff
@@ -314,8 +314,11 @@ def get_spec_handler(self, resource):
         rpath = resource['resource_path']
         root = resource.get('root', '')
         root = self.root_map.get(root, root)
-        if root:
-            rpath = os.path.join(root, rpath)
+        if os.path.isabs(rpath):
+            rpath.lstrip('/')
```
Comment (klauer) [inline filestore/fs.py]: Strings are immutable - you need to do `rpath = rpath.lstrip('/')`

**Comment by CJ-Wright** [comment]
Should me mandate that `root` be an absolute path?

**Comment by tacaswell** [comment]
Is that root actually in the data base are is that due to your use of root_map?

**Comment by CJ-Wright** [comment]
Yes it's in the db. This is most likely from the use of new root in the export process.

**Comment by tacaswell** [comment]
Yeah, that is probably a bug in the export code.

**Comment by CJ-Wright** [comment]
Hmm ok. Do you still want the PR?

**Comment by tacaswell** [comment]
If this PR does anything it should blow up in a more informative way

**Comment by danielballan** [comment]
I'm wary of hard-coding `/` anywhere, as I recently ran into problems with filestore + Windows IOCs. I second Tom's comment and encourage `os.sep` in place of `/`.

**Comment by CJ-Wright** [comment]
I guess we could also try to fix this with pathlib

**Comment by tacaswell** [comment]
The real fix here is to fix the database.

**Comment by CJ-Wright** [comment]
That too, is there a plan to fix the xpd filestore to bring it up to the new code?

**Comment by danielballan** [comment]
Good point, @tacaswell. Would a database fix solve your problem @CJ-Wright? It's not at the top of our list, but we could try to make it happen for the new cycle.

Please open an issue against the databroker if code changes are still needed. Otherwise, we can handle a database fixing task via email.

**Comment by CJ-Wright** [comment]
Yes I think a fix of the database would solve the problem.

**Pull request closed without merging**